lib/options-handler.ts

```diff
@@ -29,14 +29,96 @@ import fs from 'fs-extra';
 import semverParser from 'semver';
 import _ from 'underscore';
 
+import {LanguageKey} from '../types/languages.interfaces';
+
 import {logger} from './logger';
-import {getToolTypeByKey} from './tooling';
+import {CompilerProps} from './properties';
+import {PropertyGetter, PropertyValue} from './properties.interfaces';
+import {Source} from './sources';
+import {BaseTool, getToolTypeByKey} from './tooling';
+import {ToolTypeKey} from './tooling/base-tool.interface';
 import {asSafeVer, getHash, splitArguments, splitIntoArray} from './utils';
 
+// TODO: There is surely a better name for this type. Used both here and in the compiler finder.
```

Comment:
Yeah...no ideas on better names...but tihs is OK by me!

Reply:
As long as it's good for now! We can revisit when we get the types more solidified :)